src/Http/Middleware/SsoLogger.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Northwestern\SysDev\SOA\Http\Middleware;
+
+use http\ENV;
+use Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode as Middleware;
+use Auth;
+use App\Access;
+use Closure;
+use Illuminate\Support\Facades\Log;
+
+class SsoLogger extends Middleware
+{
+    protected $max_logs;
+
+    public function __construct() { $this->max_logs = ENV('SSO_LOG_MAX_ENTRIES',1000000); }
+
+    public function handle($request, Closure $next, ...$guards)
+    {
+        $controller = $request->route()->controller;
+
+        if (class_basename($controller) == "WebSSOController") {
+            // They're doing auth right now so we'll call it an anonymous request (no netid)
+            $this->log_request(true, $request);
+            return $next($request);
+        } else {
+            if (Auth()->user() == null) {
+                $this->log_request(true, $request);
+            } else {
+                $this->log_request(false, $request);
+            }
+        }
+
+
+        if (sizeof(Access::all()) > $this->max_logs) {
```

Comment:
```suggestion
        if (Access::count() > $this->max_logs) {
```

`all()` returns a collection with all the rows -- if you have 10,000 rows, you'd load 10,000 Eloquent models and then count them up. That would be a very noticeable performance hit.

The `count()` method will just do a `SELECT count(*) FROM table`, which will be faster.